compiler/src/dmd/backend/cod2.d

```diff
@@ -2408,7 +2408,10 @@ void cdcond(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
             codelem(cdb,e1,&retregs,false);
             const reg = findreg(retregs);
 
-            if (v1 == 0 && v2 == ~cast(targ_size_t)0)
+            // https://issues.dlang.org/show_bug.cgi?id=23743
+            // Optimization has bugs where it negates the wrong register,
```

Comment:
Is it an optimisation?

Reply:
I was able to reproduce with -O turned off but more generally I'm skeptical of these parts of the compiler because they're clearly not well tested (i.e. it's 2023 and dmd is getting sign extension wrong)

Reply:
It's not an optimization in the sense that it doesn't require the `-O` switch. It's still a peephole optimization because it generates smaller code for a specific scenario (`condition ? -1 : 0`) as opposed to the more general scenario (the else branch)